### DISCUSSION TITLE
不具合: 下書きを保存した直後の最終編集時間がマイナス秒になる

### DISCUSSION THREAD

**Discussion by uh-zz** [increments/qiita-discussions#578] [要望・アイデア・不具合]
> [!NOTE]
毎年アドベントカレンダーで楽しませていただいてます 👍 

## 概要

下書きを保存して直後の最終編集時間がスクショのとおり、マイナスになっているので、もしかしたら不具合かもしれないと思い報告させていただきます

<img width="441" alt="image" src="https://github.com/increments/qiita-discussions/assets/47747828/40ddb8d5-e6bc-4ada-88ea-d2ae631977b2">

## 提案

`0 sec < current_time - updated_at < 1 sec`の場合は、`now` みたいな表示にしてもよさそうと思いましたのでコメントさせていただきます 👋

**Comment by tanaka-asami415** [comment]
@uh-zz 
ご報告 & Advent Calendarにご参加いただき、ありがとうございます。

我々の環境では再現しなかったのですが、不具合がないか確認してみます。
ご提案いただいた内容も参考にさせていただきます🙇

**Reply by yoshinoyuya** [reply]
@tanaka-asami415 
憶測なんですが、updated_time(サーバー)、now_time(ローカル)となって、
以下のような実装のイメージなんですが
ローカルの時間がずれていてマイナスの値になったとかじゃないでしょうか
~~~
diff = now_time(ローカル) - updated_time(サーバー)
print(diff)
~~~

一分ほどPCの時間をずらした時の写真です。↓
![image](https://github.com/increments/qiita-discussions/assets/85545443/16ee3140-9d6d-4cb0-ad3d-30ac04fc3156)

**Reply by tanaka-asami415** [reply]
@yoshinoyuya 
検証までしていただき、ありがとうございます！✨🙇
修正の際に参考にさせていただきます！

**Reply by P-SiZK** [reply]
@uh-zz @yoshinoyuya 
大変お待たせいたしました :bow:
本件について修正を行いました！

サーバーとクライアントの時間のズレによりマイナス表示になってしまっていたため、
一定時間以内の差の場合は `now` と表示するように変更しました。

よろしければご確認ください :pray:
